cloud-sql-java-functions-benchmark/src/main/java/functions/Main.java

```diff
@@ -0,0 +1,215 @@
+package functions;
+
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import io.opencensus.common.Scope;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.Status;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.sql.DataSource;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class Main implements HttpFunction {
+
+  private static final String PROJECT_ID = System.getenv("PROJECT_ID");
+  private static final String CONNECTION_NAME = System.getenv("CLOUD_SQL_CONNECTION_NAME");
+  private static final String DB_NAME = System.getenv("DB_NAME");
+  private static final String DB_USER = System.getenv("DB_USER");
+  private static final String DB_PASSWORD = System.getenv("DB_PASS");
+  private static final Logger LOGGER = Logger.getLogger(Main.class.getName());
+
+  private static class LazyInitializer {
+    private static final Tracer TRACER = Tracing.getTracer();
+    static {
+      try {
+        // Set up StackDriver
+        StackdriverTraceExporter.createAndRegister(
+            StackdriverTraceConfiguration.builder()
+                .setProjectId(PROJECT_ID)
+                .build());
+
+      } catch (java.io.IOException e) {
+        Main.LOGGER.log(Level.WARNING, "Warning: Could not set up Stackdriver Trace", e);
+      }
+
+      // Trace every request
+      TraceConfig traceConfig = Tracing.getTraceConfig();
+      traceConfig.updateActiveTraceParams(
+          traceConfig.getActiveTraceParams().toBuilder().setSampler(Samplers.alwaysSample())
+              .build());
+    }
```

Comment:
I don't think this is quite right - I think you want to move all this into a function and assign it to a field, which is then called (somewhat ironically the example I previously linked looks actually more like the [Singleton pattern](https://en.wikipedia.org/wiki/Singleton_pattern)...):
```java
public final class SingletonTracer {

    private Singleton() {}

    private static final Tracer INSTANCE = initTracer();

    public static Tracer get() {
        return INSTANCE;
    }
}
```

Reply:
As a tangent, the benefits of this over the Initialization-on-demand holder is that it'll be initialized immodestly, vs the first time we call the get function. This is preferable since we use the Tracer every time.

Reply:
Got it. I was a bit confused before but that makes sense. Fixed.